### PR TITLE
on Long Press: Edit, Delete and Duplicate

### PR DESCRIPTION
**Pull request by Krunal-375**
# Fixes issue #129 

Hey @a-mabe I have added long press functionality which allow user to edit, delete and duplicate workout kindly review it.
## Here is demo video:

https://github.com/a-mabe/OpenHIIT/assets/72144846/d779cc0e-03ae-442e-b003-508817280755

### CODE REVIEW

**Review by a-mabe — CHANGES_REQUESTED** [review]
Thank you for the contribution @Krunal-375!

The duplicate workout functionality works great, and the video of the functionality is appreciated.

I have provided feedback on the use of onLongPress as well as an alternative. I had a couple of testers find it difficult to continue using the drag and drop workout reorder feature due to the conflicting gestures. However I do like the idea of the onLongPress menu, I am not sure how to better adjust the sensitivity of the onLongPress and onDrag gestures.

File: lib/duplicate_feature/duplicate_workout.dart [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import 'package:openhiit/workout_data_type/workout_type.dart';
```
[the remark this answers is not in the document]
Comment (Krunal-375) [reply lib/duplicate_feature/duplicate_workout.dart]: I have removed the duplicate_feature folder

File: lib/main.dart [hunk 2]
```diff
@@ -154,15 +176,91 @@ class _MyHomePageState extends State<MyHomePage> {
...
+            /// The key is used to control or manage the state for this specific widget.
+            key: Key('${workout.workoutIndex}'),
+
+            /// This event is triggered when the user taps down on the screen.
+            /// It calls the _getTapPosition method to update the position of the tap event.
+            onTapDown: (details) => _getTapPosition(details),
+
+            /// This event is triggered when the user presses and holds on the screen.
+            /// It opens a context menu with options to edit, delete, or duplicate the workout.
+            onLongPress: (){
```
[the remark this answers is not in the document]
Comment (Krunal-375) [reply lib/main.dart]: I have made the following changes

File: lib/main.dart [hunk 3]
```diff
@@ -154,15 +153,14 @@ class _MyHomePageState extends State<MyHomePage> {
         /// For each workout in the returned DB data snapshot.
         ///
         for (final workout in snapshot.data)
+          /// This is a GestureDetector widget that is used to handle different types of touch events.
```
Comment (a-mabe) [inline lib/main.dart:156]: Please remove the comment referencing the gesture detector as it is no longer being used

File: lib/main.dart [hunk 4]
```diff
@@ -154,15 +153,14 @@ class _MyHomePageState extends State<MyHomePage> {
...
           TimerListTile(
-            key: Key(
-                '${workout.workoutIndex}'), // Unique key for each list item.
```
Comment (a-mabe) [inline lib/main.dart:159]: Please add back these comments in `main.dart`

**Review by a-mabe — CHANGES_REQUESTED** [review]
Thank you for making those changes @Krunal-375. I found an old comment that needed fixed, but otherwise this is working great, thank you!

**Review by a-mabe — APPROVED** [review]

**Pull request merged**